### PR TITLE
SLIM-845_Replace_SoapUI_project_with_Java_Clients_for_SmartMeterAdhoc

### PR DESCRIPTION
**Pull request by riannefijn**

### CODE REVIEW

**Comment by riannefijn** [comment]
retest this please

File: cucumber-tests-platform-smartmetering/src/test/java/com/alliander/osgp/cucumber/platform/smartmetering/glue/steps/ws/smartmetering/smartmeteringadhoc/AssociationLnObjects.java [hunk 1]
```diff
@@ -7,62 +7,64 @@
...
+                .fromScenarioContext();
+        final GetAssociationLnObjectsResponse response = this.responseClient.getResponse(asyncRequest);
+
+        assertEquals("Response should be OK", OsgpResultType.OK, response.getResult());
+        assertNotNull("Response should contain an AssociationLnList", response.getAssociationLnList());
+        assertNotNull("AssociationLnList shoudl have at least one entry",
+                response.getAssociationLnList().getAssociationLnListElement().get(0));
+
+        final AssociationLnListElement element = response.getAssociationLnList().getAssociationLnListElement().get(0);
+        assertNotNull("AccessRights should be present", element.getAccessRights());
```
Comment (robindenadel) [inline cucumber-tests-platform-smartmetering/src/test/java/com/alliander/osgp/cucumber/platform/smartmetering/glue/steps/ws/smartmetering/smartmeteringadhoc/AssociationLnObjects.java]: Check for AttributeId as well

File: cucumber-tests-platform-smartmetering/src/test/java/com/alliander/osgp/cucumber/platform/smartmetering/support/ws/smartmetering/adhoc/SmartMeteringAdHocRequestClient.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+import com.alliander.osgp.adapter.ws.schema.smartmetering.common.AsyncResponse;
+import com.alliander.osgp.cucumber.platform.smartmetering.support.ws.smartmetering.SmartMeteringBaseClient;
+import com.alliander.osgp.shared.exceptionhandling.WebServiceSecurityException;
+import com.alliander.osgp.shared.infra.ws.DefaultWebServiceTemplateFactory;
+
+@Component
+public class SmartMeteringAdHocRequestClient<T extends AsyncResponse, V> extends SmartMeteringBaseClient {
+
+    @Autowired
+    private DefaultWebServiceTemplateFactory smartMeteringAdHocWstf;
```
Comment (robindenadel) [inline cucumber-tests-platform-smartmetering/src/test/java/com/alliander/osgp/cucumber/platform/smartmetering/support/ws/smartmetering/adhoc/SmartMeteringAdHocRequestClient.java]: Do not use abbreviations like Wstf in variable names. Rename smartMeteringAdHocWstf.  Maybe webServiceTemplateFactory will do

File: cucumber-tests-platform-smartmetering/src/test/java/com/alliander/osgp/cucumber/platform/smartmetering/support/ws/smartmetering/adhoc/SmartMeteringAdHocResponseClient.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+import com.alliander.osgp.adapter.ws.schema.smartmetering.common.AsyncRequest;
+import com.alliander.osgp.cucumber.platform.smartmetering.support.ws.smartmetering.SmartMeteringBaseClient;
+import com.alliander.osgp.shared.exceptionhandling.WebServiceSecurityException;
+import com.alliander.osgp.shared.infra.ws.DefaultWebServiceTemplateFactory;
+
+@Component
+public class SmartMeteringAdHocResponseClient<T, V extends AsyncRequest> extends SmartMeteringBaseClient {
+
+    @Autowired
+    private DefaultWebServiceTemplateFactory smartMeteringAdHocWstf;
```
Comment (robindenadel) [inline cucumber-tests-platform-smartmetering/src/test/java/com/alliander/osgp/cucumber/platform/smartmetering/support/ws/smartmetering/adhoc/SmartMeteringAdHocResponseClient.java]: Do not use abbreviations like Wstf in variable names.

File: cucumber-tests-platform-smartmetering/src/test/java/com/alliander/osgp/cucumber/platform/smartmetering/support/ws/smartmetering/adhoc/SynchronizeTimeRequestFactory.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+    private SynchronizeTimeRequestFactory() {
+        // Private constructor for utility class
+    }
+
+    public static SynchronizeTimeRequest fromParameterMap(final Map<String, String> parameters) {
+        final SynchronizeTimeRequest request = new SynchronizeTimeRequest();
+        request.setDeviceIdentification(parameters.get(PlatformSmartmeteringKeys.DEVICE_IDENTIFICATION));
+        request.setSynchronizeTimeRequestData(SynchronizeTimeRequestDataFactory.fromParameterMap(parameters));
+        ;
```
Comment (robindenadel) [inline cucumber-tests-platform-smartmetering/src/test/java/com/alliander/osgp/cucumber/platform/smartmetering/support/ws/smartmetering/adhoc/SynchronizeTimeRequestFactory.java]: remove line

**Pull request merged**